applogic/src/util/file_ring_buffer.rs

```diff
@@ -0,0 +1,220 @@
+// SPDX-FileCopyrightText: 2025 Phoenix R&D GmbH <hello@phnx.im>
```

Comment:
The amount of direct slice access and unchecked additions and subtractions in the code make me a bit nervous. It looks sound in most places, but it all seems a bit error-prone. How much worse would it look if we used `.get()` and checked operations in most places?

Reply:
I will change the slice accesses to a check version, except for the slice ranged access. I will also add a small `quickcheck` testing the ring buffer better.